src/serde/reader.rs

```diff
@@ -90,19 +90,15 @@ impl fmt::Display for DeserializeError {
                 write!(fmt, "InvalidEncoding: {}", ib),
             DeserializeError::SizeLimit =>
                 write!(fmt, "SizeLimit"),
-            DeserializeError::Serde(ref s) =>
-                s.fmt(fmt),
+            DeserializeError::Custom(ref s) =>
+                write!(fmt, "Custom Error {}", s),
```

Comment:
I would prefer to just format the message like before - `s.fmt(fmt)`. I don't think "Custom Error" adds any clarity or context.